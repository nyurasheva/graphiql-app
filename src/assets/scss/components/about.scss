/* about */

.about {
  width: 100%;
  background: $background;
  z-index: 8;

  &__title {
    color: $button;
    padding: 70px 0 100px;
  }

  &__content {
    background: $text-color;
    width: 100%;
    padding: 0 0 5px;

    p {
      color: $background;
      text-align: left;
    }
  }

  &__item {
    @include flex($justify: stretch);
    width: calc(75% + 10px);
    height: auto;
    margin: 0 auto 90px;
    position: relative;
    border: 1px solid #ffffff40;
    background: #1e1e1e;
    transition: $transition;

    @include media('xs') {
      flex-direction: column;
    }

    &:hover {
      background: #454545;
      cursor: pointer;

      .about__number {
        top: -40px;

        @include media('l') {
          top: -26px;
        }
      }
    }
  }

  &__images,
  &__descriptions {
    @include flexColumn($align: stretch);
    text-align: left;
    opacity: 0;
    transition:
      opacity 1s ease,
      transform 1s ease;

    &.visible {
      opacity: 1;
      transform: none;
    }
  }

  &__descriptions {
    width: 60%;
    padding: 0 40px 20px 20px;
    gap: 25px;
    text-shadow: 2px 2px 2px rgba(0, 0, 0, 0.5);
    transform: translateY(50%) translateX(50%);

    @include media('l') {
      gap: 8px;
      padding: 10px;
    }

    @include media('xs') {
      width: 100%;
      padding-top: 20px;
    }
  }

  &__subtitle {
    @include line();
    color: $button-second;

    @include media('l') {
      font-size: 1.6rem;
    }
  }

  &__images {
    width: 40%;
    transform: translateY(-50%) translateX(-50%);

    @include media('xs') {
      width: 100%;
    }
  }

  &__img {
    height: auto;
    width: 100%;
    background-size: cover;
    background-position: top center;
    box-shadow: 0px 3px 4px 0px #00000033;
    position: relative;
    top: -50px;
    left: -50px;

    @include media('m') {
      left: -25px;
    }

    @include media('xs') {
      top: 0;
      left: 0;
    }
  }

  &__number {
    @include font($family: $font-family-title, $size: 1.4rem, $weight: bold);
    position: absolute;
    width: 2em;
    height: 2em;
    padding: 10px;
    top: -20px;
    right: 22px;
    color: $links;
    border: 2px solid $links;
    border-radius: 50%;
    text-align: center;
    transition: $transition;

    @include media('l') {
      font-size: 1.1rem;
      width: 1.6em;
      height: 1.6em;
      padding: 5px;
      top: -13px;
      right: 10px;
    }

    @include media('s') {
      padding: 4px;
    }
  }

  &__svg {
    width: 100%;
    margin-top: -1px;
  }
}
