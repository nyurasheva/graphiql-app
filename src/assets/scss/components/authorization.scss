/* authorization */

.authorization {
  @include flex($justify: space-between, $align: baseline);
  width: 100%;
  padding: 20px 0;
  gap: 20px;

  @include media('m') {
    @include flexColumn($align: stretch);
    gap: 35px;
  }

  &__container,
  &__information {
    flex: 1;
    text-align: left;
  }

  &__container {
    h1 {
      text-align: left;
      color: #ffe1f1;
    }

    .form {
      @include flexColumn($align: stretch);
    }

    .label {
      @include flexColumn($align: start);
      margin-bottom: 25px;
      position: relative;
      width: auto;
      color: $text-color;
      float: left;
      padding: 6px 0;

      span {
        color: #9ffffb;
      }

      input {
        width: 80%;
        min-width: 300px;

        @include media('xxs') {
          width: 75%;
          min-width: 200px;
        }
      }

      &.error {
        span {
          color: $error;

          @include media('l') {
            font-size: 0.8rem;
          }
        }

        input {
          border-color: $error;
        }
      }
    }

    .button {
      margin-top: 15px;
    }
  }

  &__information {
    h4 {
      @include font($family: $font-family-title, $size: 1.4rem);
      padding-bottom: 10px;
      margin-bottom: 10px;
      color: #bdc3c7;
    }

    p {
      padding: 15px 0;
    }
  }

  &__login {
    margin-top: 15px;
    @include font($family: $font-family-title, $color: $background);
  }
}

.error-message {
  top: 70px;
  position: absolute;
  color: $important;
  font-size: 0.8rem;

  @include media('xxs') {
    font-size: 10px;
  }
}
