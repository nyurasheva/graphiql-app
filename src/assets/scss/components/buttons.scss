/* buttons */

.button {
  background: $links;
  color: $background;
  display: inline-flex;
  align-items: center;
  justify-content: center;
  width: max-content;
  min-width: $button-size-40;
  height: $button-size-40;
  white-space: normal;
  line-height: 1.2;
  padding: 5px 15px;
  border: none;
  background-clip: border-box;
  letter-spacing: normal;
  text-transform: none;
  border-radius: 40px;
  transition: $transition;

  &:hover {
    color: $background;
    background: $button-hover;
    cursor: pointer;
    box-shadow:
      0 0 0 0.0625rem #241c15,
      0 0.375em 0 0 #241c15;
    transform: translateY(calc(-1 * 0.375em));
  }
}

.button-second {
  background: $background-gradient;
  transition: $transition;

  &:hover {
    background: linear-gradient(135deg, #00cac2 30%, #2f90d2 100%);
  }

  &.confirm {
    min-width: 161px;

    span {
      width: 100%;
      height: 100%;
      display: flex;
      justify-content: center;
      align-items: center;
      position: relative;
    }
  }
}

.button-transparent {
  background: transparent;
  color: $links;
  border: 2px solid $links;

  &:hover {
    border: 2px solid transparent;
  }
}
