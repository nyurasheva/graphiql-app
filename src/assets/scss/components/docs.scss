aside {
  position: fixed;
  z-index: 20;
  right: 0;
  bottom: 0;
  top: 0;
  left: 100%;
  background: #272b33;
  padding: 0.5rem 1rem;
  transition: all 1s ease;

  .docs {
    &-container {
      height: 100%;
      width: 100%;
      padding-top: 10px;
      overflow-y: auto;
    }

    &-button {
      position: absolute;
      left: 0;
      top: 50%;
      background: $links;
      border-bottom-left-radius: 0;
      border-bottom-right-radius: 0;
      transform: rotate(270deg) translateY(-178%);
      padding: 1rem 3rem;
      color: $background;
    }

    &-details {
      @include flexColumn(flex-start, flex-start);
      row-gap: 1rem;
      padding: 1rem;
      color: #9ffffb;

      &__type {
        @include flex();

        & span {
          color: #98c379;
          padding-left: 5px;
        }
      }

      &__description {
        @include flexColumn($align: baseline);

        & span {
          color: #d6d1e8;
          text-align: left;
        }
      }
    }
  }
}

.openAside {
  left: 60%;
  transition: all 1s ease;
}
