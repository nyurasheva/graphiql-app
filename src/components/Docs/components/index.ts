export { default as FieldSchema } from './FieldSchema';
export { default as DetailedField } from './DetailedField';
export { default as NextField } from './NextField';
