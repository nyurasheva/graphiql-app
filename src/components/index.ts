export { default as LoginForm } from './LoginForm';
export { default as Logo } from './Logo';
export { default as RegistrationForm } from './RegistrationForm';
export { default as Docs } from './Docs';
