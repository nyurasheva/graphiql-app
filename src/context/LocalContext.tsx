import React, {
  createContext,
  useState,
  useEffect,
  useContext,
  ReactNode,
} from 'react';
import { LocalizationContextType, Translation } from './types';
import { translationsData } from './LocalData';

const LocalizationContext = createContext<LocalizationContextType | undefined>(
  undefined
);

type LocalizationProviderProps = {
  children: ReactNode;
};

export const LocalizationProvider: React.FC<LocalizationProviderProps> = ({
  children,
}) => {
  const [language, setLanguage] = useState<string>('en');
  const [translations, setTranslations] = useState<Translation>({});

  const changeLanguage = (newLanguage: string) => {
    setLanguage(newLanguage);
  };

  useEffect(() => {
    setTranslations(translationsData[language]);
  }, [language]);

  return (
    <LocalizationContext.Provider
      value={{ language, translations, changeLanguage }}
    >
      {children}
    </LocalizationContext.Provider>
  );
};

export const Localization = () => {
  const context = useContext(LocalizationContext);
  if (!context) {
    throw new Error('Localization must be used within a LocalizationProvider');
  }
  return context;
};
