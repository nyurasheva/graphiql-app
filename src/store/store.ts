import { combineReducers, configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/query';
import userReducer from './slice/userSlice';
import requestReducer from './slice/requestSlice';

const reducer = combineReducers({
  user: userReducer,
  request: requestReducer,
});

const store = configureStore({
  reducer,
});

setupListeners(store.dispatch);

export default store;
