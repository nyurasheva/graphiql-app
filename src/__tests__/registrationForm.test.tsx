import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { RegistrationForm } from '../components';
import { LocalizationProvider } from '../context/LocalContext';
import store from '../store/store';

describe('registrationForm component', () => {
  it('should match registrationForm snapshot', () => {
    const { asFragment } = render(
      <LocalizationProvider>
        <Provider store={store}>
          <BrowserRouter>
            <RegistrationForm />
          </BrowserRouter>
        </Provider>
      </LocalizationProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });

  jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom'),
    useNavigate: jest.fn(),
  }));

  // it('should change registrationForm`s input', async () => {
  //   const navigateMock = jest.fn();
  //   (useNavigate as jest.Mock).mockReturnValue(navigateMock);

  //   render(
  //     <LocalizationProvider>
  //       <Provider store={store}>
  //         <BrowserRouter>
  //           <RegistrationForm />
  //         </BrowserRouter>
  //       </Provider>
  //     </LocalizationProvider>
  //   );

  //   const input = await screen.findByPlaceholderText('John');
  //   const email = await screen.findByPlaceholderText('email@gmail.com');
  //   const password = document.getElementById('password') as HTMLInputElement;
  //   const confirmPassword = document.getElementById(
  //     'confirmPassword'
  //   ) as HTMLInputElement;

  //   fireEvent.change(input, { target: { value: 'qqq' } });
  //   fireEvent.change(email, { target: { value: 'test@gmail.com' } });
  //   fireEvent.change(password, { target: { value: 'qwertyQ!1' } });
  //   fireEvent.change(confirmPassword, { target: { value: 'qwertyQ!1' } });

  //   const form = document.querySelector('.form') as HTMLFormElement;
  //   fireEvent.submit(form);

  //   expect(navigateMock).toHaveBeenCalled();
  // });
});
