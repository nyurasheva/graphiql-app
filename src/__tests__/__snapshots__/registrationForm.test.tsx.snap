// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`registrationForm component should match registrationForm snapshot 1`] = `
<DocumentFragment>
  <form
    class="form"
  >
    <label
      class="label "
      for="firstName"
    >
      <span>
        Name:
      </span>
      <input
        id="firstName"
        name="firstName"
        placeholder="John"
        type="text"
      />
    </label>
    <label
      class="label "
      for="email"
    >
      <span>
        Email:
      </span>
      <input
        id="email"
        name="email"
        placeholder="email@gmail.com"
        type="text"
      />
    </label>
    <label
      class="label "
      for="password"
    >
      <span>
        Password (minimum 8 symbols, including a digit, letter, and special character)
      </span>
      <input
        id="password"
        name="password"
        type="password"
      />
    </label>
    <label
      class="label "
      for="confirmPassword"
    >
      <span>
        Confirm Password:
      </span>
      <input
        id="confirmPassword"
        name="confirmPassword"
        type="password"
      />
    </label>
    <button
      class="button button-second"
      type="submit"
    >
      Registration
    </button>
  </form>
</DocumentFragment>
`;
